### ISSUE TITLE
Cannot "Package" service fabric application anymore

### ISSUE THREAD

**Issue by Jaans** [microsoft/service-fabric#360]
I'm having a strange problem that I'm unable to resolve.
I used to be able to "`Package`" and/or "`Publish`" my Service Fabric project without issue, but now, after updating VS 2017 15.9.14 and latest SF DEV SDK updates, I'm no longer able to "`Package`" (or "Publish" which does an implicit "Package" first anyway).

The error message is: 
> An error occurred while attempting to package the application.
> Value cannot be null. Parameter name: path1

Problem is there is no additional error information (other than the screenshot below) available anywhere (Not in detailed level build log, event logs, command line, nothing).

![image](https://user-images.githubusercontent.com/3510981/61220279-5581ab00-a759-11e9-8c9b-2f0cc8f3cde8.png)

Running Latest SF 6.5 and SDK 6.5.641/3.4641

I might be falsely assuming that the "Build" step is working correctly too, but it doesn't give any errors. I have even stripped everything down to one stateless service project reference and minimal manifest, but it makes no difference.

Please help.

**Comment by Jaans** [comment]
Some additional information.
The `Microsoft Azure Service Fabric Tools for Visual Studio` version is **2.5**:
![image](https://user-images.githubusercontent.com/3510981/61268303-5d376300-a7de-11e9-81ee-adea6d91bd9b.png)


I have another installation of VS 2017 on my laptop where **this issue is not happening** with the exact same source code. This version has Visual Studio 2017 at **v15.9.12** and Microsoft Azure Service Fabric Tools for Visual Studio is at **v2.4**
To be clear, _this version still works._

The problem is I cannot rollback to that version with the way that Visual Studio Installer works. According to [this guide](https://docs.microsoft.com/en-us/visualstudio/productinfo/installing-an-earlier-release-of-vs2017) I need to uninstall Visual Studio 2017 completely and use a specific version installer. That takes me to VS 2017 **v15.8.9** (which has Service Fabric Tools at **v2.3**).

Worse, the installed doesn't allow me to upgrade to v15.9.12 - it only allows me to jump to the latest v15.9.14.

With v15.8.9 the error message from this issue isn't showing, but it's also not outputting the referenced project binary build outputs to the pkg folder of the Service Fabric solution.

Reinstalling Visual Studio and even downgrading the local dev instance of the SF SDK doesn't do anything. I'm out of options and my Google Fu is now depleted as there is nothing I can find help resolve the issue. 

Please help.

**Comment by Jaans** [comment]
I've managed to get the old version of VS 2017 v15.8.9 working again, after completely removing the local development cluster runtime and SDK and installing an old version v6.4.664.9590 and v3.3.644.9590 respectively. 

So now I'm stuck on a much older version because VS upgrade only jumps to latest. 
I did test upgrading to latest version of VS 2017 v15.9.14 that the issue returns after getting it to work on the old version and can confirm that the issue is still there. Now to uninstall it all again and start over...

**Comment by Jaans** [comment]
Today we moved to Visual Studio 2019 v16.3.1 with all the latest updates and the issue remains. The error presents a little differently, but I still cannot "Package" our ServiceFabric project:

![image](https://user-images.githubusercontent.com/3510981/65816987-74b2b380-e245-11e9-9c76-27e1a04958bd.png)

So very sad :-(

**Comment by Sashwat** [comment]
> Today we moved to Visual Studio 2019 v16.3.1 with all the latest updates and the issue remains. The error presents a little differently, but I still cannot "Package" our ServiceFabric project:
> 
> ![image](https://user-images.githubusercontent.com/3510981/65816987-74b2b380-e245-11e9-9c76-27e1a04958bd.png)
> 
> So very sad :-(

Were you able to resolve this issue? I am stuck on the same

**Comment by Jaans** [comment]
@Sashwat No, we were not able to get this resolved. Gave up waiting for any kind of response - more sadness :-(

Instead we now run our own custom PowerShell scripts to package up our build output and prepare it for deployment. It is quite a bit of work to get it working just right for us, but it is so much faster and better for us now. 

Sure, it doesn't do some of the automatic versioning number and validation bits that VS added, but we just do those by hand via the script. 

Everything that VS does to build, package and deploy can be done with the Service Fabric PowerShell commands and some good scripting. This might be of interest to get you started to what VS does under the hood: https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-package-apps
